src/main/java/com/netflix/imflibrary/st2067_2/Composition.java

```diff
@@ -0,0 +1,1154 @@
+     *
+     * @return string representing the object
+     */
+    public String toString()
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append(String.format("======= Composition : %s =======%n", this.uuid));
+        sb.append(this.editRate.toString());
+        return sb.toString();
+    }
+
+    /**
+     * A method that confirms if the inputStream corresponds to a Composition document instance.
+     * @param resourceByteRangeProvider corresponding to the Composition XML file.
+     * @return a boolean indicating if the input file is a Composition document
+     * @throws IOException - any I/O related error is exposed through an IOException
+     */
+    public static boolean isCompositionPlaylist(ResourceByteRangeProvider resourceByteRangeProvider) throws IOException{
+        try(InputStream inputStream = resourceByteRangeProvider.getByteRangeAsStream(0, resourceByteRangeProvider.getResourceSize()-1);)
+        {
+            DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+            documentBuilderFactory.setNamespaceAware(true);
+            DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+            Document document = documentBuilder.parse(inputStream);
+
+            //obtain root node
+            NodeList nodeList = null;
+            for(String cplNamespaceURI : Composition.supportedCPLSchemaURIs) {
+                nodeList = document.getElementsByTagNameNS(cplNamespaceURI, "CompositionPlaylist");
+                if (nodeList != null
+                        && nodeList.getLength() == 1)
+                {
+                    return true;
+                }
+            }
+        }
+        catch(ParserConfigurationException | SAXException e)
+        {
+            return false;
+        }
+
+        return false;
+    }
+
+    /**
+     * Getter for the composition edit rate as specified in the Composition XML file
+     * @return the edit rate associated with the Composition
+     */
+    public EditRate getEditRate()
+    {
+        return this.editRate;
+    }
+
+    /**
+     * Getter for the virtual track map associated with this Composition
+     * @return {@link java.util.Map Map}&lt;{@link java.util.UUID UUID},{@link Composition.VirtualTrack VirtualTrack}&gt;. The UUID key corresponds to VirtualTrackID
+     */
+    public Map<UUID, ? extends VirtualTrack> getVirtualTrackMap()
+    {
+        return Collections.unmodifiableMap(this.virtualTrackMap);
+    }
+
+    /**
+     * Getter for the UUID corresponding to this Composition document
+     * @return the uuid of this Composition object
+     */
+    public UUID getUUID()
+    {
+        return this.uuid;
+    }
+
+    /**
+     * Getter for the CompositionPlaylistType object model of the Composition defined by the st2067-3 schema.
+     * @return the composition playlist type object model.
+     */
+    public JAXBElement getCompositionPlaylistTypeJAXBElement(){
```

Comment:
Wouldnt it be better if this was an internal method of the class? e.g., that would help hide the JAXB aspects of our implementation